src/learner.cc

```diff
@@ -345,141 +481,148 @@ class LearnerImpl : public Learner {
     }
   }
 
-  void CheckDataSplitMode() {
-    if (rabit::IsDistributed()) {
-      CHECK(tparam_.dsplit != DataSplitMode::kAuto)
-        << "Precondition violated; dsplit cannot be 'auto' in distributed mode";
-      if (tparam_.dsplit == DataSplitMode::kCol) {
-        // 'distcol' updater hidden until it becomes functional again
-        // See discussion at https://github.com/dmlc/xgboost/issues/1832
-        LOG(FATAL) << "Column-wise data split is currently not supported.";
+  void ConfigureNumFeatures(LearnerAttributes* attr) {
```

Comment:
Can we remove this in the near future and use DMatrix as the source of truth?

Reply:
Once the JVM PR is merged.